dali/pipeline/util/batch_rng.h

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_UTIL_BATCH_RNG_H_
+#define DALI_PIPELINE_UTIL_BATCH_RNG_H_
+
+#include <random>
+#include <vector>
+
+#include "dali/core/span.h"
+
+namespace dali {
+
+template <typename RNG = std::mt19937>
+class BatchRNG {
+ public:
+  BatchRNG(int64_t seed, int batch_size, int state_size = 4) : seed_(seed) {
```

Comment:
A comment that `state_size` helps us avoid seed collision and generating identical sequences for two samples would be welcome. Otherwise it may seem a bit obscure to the reader.

Reply:
Done